Clarify using of the `ALTER SERVER` options (#2551)
neo4j/docs-operations: modules/ROOT/pages/clustering/servers.adoc

As with the `DEALLOCATE DATABASES FROM SERVER ...` command, if the alteration of a server's options renders it impossible for the cluster to satisfy one or more of the databases' topologies, then the command fails and no changes are made.

[IMPORTANT]
====
Input provided to `SET OPTIONS {...}` replaces **all** existing options, rather than being combined with them.

> Any previously set values must be specified every time you run the `ALTER SERVER` command; otherwise they will be overwritten to defaults taken from the settings `initial.server.mode_constraint`, `initial.server.allowed_databases`, `initial.server.denied_databases`, and/or `initial.server.tags`.

Reviewer: This seems misleading. It sounds like if you don't specify some of the options, we will pick up the _configured setting_ from the relevant `initial.` setting you mention. In fact, we will just set the option to whatever the appropriate 'empty' value is.

Perhaps 
```suggestion
Any previously set values must be specified every time you run the `ALTER SERVER` command; otherwise they will be overwritten to the unset values.
```

The 'initial' values for a server are only relevant when you enable the server, if you don't specify something that overrides them then.

Author: ah, interesting. In the trello, we have, it's said: 

Similarly to the example given, if they (customers) have configuration for `allowedDatabases` or `deniedDatabases` and do not explicitly set them in the options for `ALTER SERVER`, the values are overwritten to defaults.

In the 'Enabled state' section, we say:
When a server is enabled, if no OPTIONS are not provided, the default server values are taken from the settings `initial.server.mode_constraint`, `initial.server.allowed_databases`, `initial.server.denied_databases`, and/or `initial.server.tags`.

That's why I decided that defaults are taken from the `initial.` settings.
